src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceHelpers.cs

```diff
@@ -51,18 +51,20 @@ public static string GetAssemblyDisplay(Compilation compilation, IAssemblySymbol
             // This method is only used to generate a comment at the top of Metadata-as-Source documents and
             // previous submissions are never viewed as metadata (i.e. we always have compilations) so there's no
             // need to consume compilation.ScriptCompilationInfo.PreviousScriptCompilation.
+            var assemblyReference = TryGetAssemblyReference(compilation, assemblySymbol);
+            return assemblyReference?.Display ?? FeaturesResources.location_unknown;
+        }
 
-            // TODO (https://github.com/dotnet/roslyn/issues/6859): compilation.GetMetadataReference(assemblySymbol)?
-            var assemblyReference = compilation.References.Where(r =>
+        public static MetadataReference TryGetAssemblyReference(Compilation compilation, IAssemblySymbol assemblySymbol)
```

Comment:
Try* methods should return boolean values and return a value as a out variable.  Otherwise just rename this to GetAssemblyReference.

Reply:
For better or worse, this is a naming pattern we use in this repository to indicate the return value can be null. Someday (soon?) we'll be able to use nullable reference types instead. 😄